.github/workflows/ci.yml

```diff
@@ -0,0 +1,62 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Run tests
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust
+        uses: dtolnay/rust-toolchain@master
+        with:
+          # Keep this in sync with `rust-toolchain.toml`.
+          toolchain: nightly-2024-08-21
+          components: rustc-dev, llvm-tools-preview
+
+      - name: Run tests
+        run: cargo test --workspace
```

Comment:
This is lacking a whole bunch of flags in order to test everything. See bevy_quickstart.